src/lib/Server/Input/Parser/MoveLocation.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+namespace Ibexa\Rest\Server\Input\Parser;
+
+use Ibexa\Contracts\Core\Repository\LocationService;
+use Ibexa\Contracts\Core\Repository\Values\Content\Location;
+use Ibexa\Contracts\Rest\Exceptions;
+use Ibexa\Contracts\Rest\Input\ParsingDispatcher;
+use Ibexa\Rest\Input\BaseParser;
+
+final class MoveLocation extends BaseParser
+{
+    public function __construct(
+        private readonly LocationService $locationService
+    ) {
+    }
+
+    /**
+     * @phpstan-param array{
+     *     'destination': string,
+     * } $data
+     *
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\UnauthorizedException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException
+     * @throws \Ibexa\Contracts\Rest\Exceptions\Parser
+     */
+    public function parse(array $data, ParsingDispatcher $parsingDispatcher): Location
+    {
+        if (!array_key_exists('destination', $data)) {
+            throw new Exceptions\Parser("Missing 'destination' element for MoveLocationInput.");
+        }
+
+        return $this->getLocationByPath($data['destination']);
+    }
+
+    /**
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\UnauthorizedException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException
+     */
+    private function getLocationByPath(string $path): Location
+    {
+        return $this->locationService->loadLocation(
+            $this->extractLocationIdFromPath($path)
+        );
+    }
+
+    /**
+     * @throws \Ibexa\Contracts\Rest\Exceptions\Parser
+     */
+    private function extractLocationIdFromPath(string $path): int
+    {
+        $pathParts = explode('/', $path);
+
+        $locationId = (int)array_pop($pathParts);
+
+        if ($locationId === 0) {
```

Comment:
Negative integers are also invalid logic-wise. How will the application behave when it receives values like:
```
2/-1
2/1foo
```
:thinking:

Reply:
Irrelevant after https://github.com/ibexa/rest/pull/95/commits/74ebd1195d7578b25175fefb383862a1d2a2dd36